lib/wallet/test/unit/Cardano/Wallet/DB/Store/Meta/StoreSpec.hs

```diff
@@ -51,3 +79,36 @@ prop_StoreMetaLaws = withInitializedWalletProp $ \wid runQ ->
         mkStoreMetaTransactions
         (pure mempty)
         (logScale . genDeltas wid)
+
+prop_QueryLaw :: WalletProperty
+prop_QueryLaw =
+    withInitializedWalletProp $ \wid runQ ->
+        forAllM (genExpand wid arbitrary) $ \history -> do
+            runQ $ writeS (store mkQueryStoreTxMeta) history
+            forM_ (Map.keys $ relations history) $ \txId -> do
```

Comment:
```suggestion
            unknownTxId <- arbitrary
            let txIds = unknownTxId : Map.keys (relations history)
            forM_ txIds $ \txId -> do
```

In this way, we would also test the "unhappy path" where `GetOne` is applied to a transaction id that is not in the history. 🤓